### ISSUE TITLE
L036 false positive when "select distinct" is used

### ISSUE THREAD

**Issue by pwildenhain** [sqlfluff/sqlfluff#677]
## Expected Behaviour

L036 (Select targets should be on a new line unless there is only one select target) allows the `distinct` clause on the same line as the `select` clause since it's not a select target

## Observed Behaviour

test.sql
```sql
select distinct
    field_1
from
    my_table
```

```
sqlfluff lint test.sql        
== [test.sql] FAIL
L:   1 | P:   1 | L036 | Select targets should be on a new line unless there is
                       | only one select target.
```

🤔 I would wager it's a problem with the rule itself, since it seems to parse correctly
```
 sqlfluff parse test.sql       
[0](1, 1, 1)        |file:
[0](1, 1, 1)        |    statement:
[0](1, 1, 1)        |        select_statement:
[0](1, 1, 1)        |            select_clause:
[0](1, 1, 1)        |                keyword:                                      'select'   
[6](1, 1, 7)        |                whitespace:                                   ' '        
[7](1, 1, 8)        |                select_clause_modifier:
[7](1, 1, 8)        |                    keyword:                                  'distinct' 
[15](1, 1, 16)      |                [META] indent:
[15](1, 1, 16)      |                newline:                                      '\n'       
[16](1, 2, 1)       |                whitespace:                                   '    '     
[20](1, 2, 5)       |                select_target_element:
[20](1, 2, 5)       |                    column_reference:
[20](1, 2, 5)       |                        identifier:                           'field_1'  
[27](1, 2, 12)      |            newline:                                          '\n'       
[28](1, 3, 1)       |            [META] dedent:
[28](1, 3, 1)       |            from_clause:
[28](1, 3, 1)       |                keyword:                                      'from'     
[32](1, 3, 5)       |                [META] indent:
[32](1, 3, 5)       |                whitespace:                                   ' '        
[33](1, 3, 6)       |                table_expression:
[33](1, 3, 6)       |                    table_reference:
[33](1, 3, 6)       |                        identifier:                           'my_table' 
[41](1, 3, 14)      |                [META] dedent:
[41](1, 3, 14)      |    newline:                                                  '\n'  
```

## Version
sqlfluff, version 0.4.0a3
Python 3.8.6

## Configuration
```
[sqlfluff]
dialect = postgres
exclude_rules = L031,L032

[sqlfluff:indentation]
indented_joins = True

[sqlfluff:rules]
max_line_length = 115

[sqlfluff:rules:L010]
capitalisation_policy = lower

[sqlfluff:rules:L014]
capitalisation_policy = lower

[sqlfluff:rules:L030]
capitalisation_policy = lower
```

**Comment by barrywhart** [comment]
I think you're right , @pwildenhain. I did some work on this rule recently, and it loops over the columns, processing or ignoring segments based on the type. Probably it needs to ignore the `DISTINCT` keyword.

I may have time to look at this in the next week or two, but if anyone wants to grab it, feel free.

**Comment by barrywhart** [comment]
On further review, I think the rule is behaving correctly. Note the rule enforces two things:
* Single `SELECT` target: Must be on the same line as the `SELECT`
* Multiple `SELECT` targets: Each must be on its own line, separate from the `SELECT`

Therefore, the correct formatting of the test query:
```
select distinct
    field_1
from
    my_table
```

is:

```
select distinct field_1
from
    my_table
```

If you remove the word `distinct` from the test query, the desired formatting is:
select field_1
from
    my_table
```

In both cases, the **one** column should be on the same line as the `SELECT.

**Comment by pwildenhain** [comment]
Ohhh good call out -- I'll test this with multiple select targets and then get back

**Comment by barrywhart** [comment]
If it turns out there is a problem, let me know -- I had already partially implemented a fix before realizing the above.

**Comment by barrywhart** [comment]
Tentatively tagged this as "invalid" , pending feedback from @pwildenhain.

**Comment by pwildenhain** [comment]
I think this is still a bug since this query:
```sql
select distinct
    field_1,
    field_2
from
    my_table
```

produces this output
```
sqlfluff lint test.sql
== [models\test.sql] FAIL
L:   1 | P:   1 | L036 | Select targets should be on a new line unless there is
                       | only one select target.
```

**Comment by barrywhart** [comment]
Makes sense. Thanks for clarifying!

**Comment by barrywhart** [comment]
@pwildenhain: What version or git commit were you using when you saw the lint failure [above](https://github.com/sqlfluff/sqlfluff/issues/677#issuecomment-772603054)? I tried it on two recent commits and am not seeing any lint warnings:
* `509312f187d9af7bc4ba7e96aa81732ff1eb88a2` (January 31)
* `53fa134877fd5d8f886065d9c51bd233895e6d84` (February 7)

**Comment by pwildenhain** [comment]
Ah ok -- I was testing on the latest alpha release, but after testing on the most recent `master` commit, this error mesage went away.

Thanks Barry!
